packages/device/src/components/menu.component.tsx

```diff
@@ -1,115 +1,137 @@
-import { createSignal, onMount, type JSX } from 'solid-js';
-import { createResource } from 'solid-js';
+import {
+  createSignal,
+  createResource,
+  onMount,
+  type JSX,
+  type Component,
+} from 'solid-js';
+import { Device } from '../classes/device';
 import { BaseMenu, type MenuEntry } from './basemenu.component';
 
-// The menu entries. Make these talk to the host app. Also make sure to on
-// show actions and settings that are available on the current device.
-//
-// There are three types of menu entries:
-// - action: a simple action such as restart or shutdown
-// - checkbox: a checkbox with a state
-// - submenu: a submenu with children
-const entries: MenuEntry[] = [
-  {
-    name: 'Restart',
-    id: 'action1',
-    type: 'action',
-    action: () => console.log('restart'),
-  },
-  {
-    name: 'Shutdown',
-    id: 'action2',
-    type: 'action',
-    action: () => console.log('shutdown'),
-  },
-  {
-    name: 'Reboot',
-    id: 'action3',
-    type: 'action',
-    action: () => console.log('reboot'),
-  },
-  {
-    name: 'Update',
-    id: 'action4',
-    type: 'action',
-    action: () => console.log('update'),
-  },
-  {
-    name: 'Debug mode',
-    id: 'debug1',
-    type: 'checkbox',
-    state: false,
-    action: (state: boolean) => console.log('debug mode', state),
-  },
-  {
-    name: 'Debug mode2',
-    id: 'debug2',
-    type: 'checkbox',
-    state: true,
-    action: (state: boolean) => console.log('debug mode2', state),
-  },
-  {
-    name: 'Advanced options',
-    id: 'advanced',
-    type: 'submenu',
-    action: (state: boolean) => console.log('Advanced options', state),
-    children: [
-      {
-        name: 'Debug mode3',
-        id: 'debug3',
-        type: 'checkbox',
-        state: true,
-        action: (state: boolean) => console.log('debug mode3', state),
-      },
-      {
-        name: 'Launch interstellar rocket',
-        id: 'action5',
-        type: 'action',
-        action: () => console.log('launch interstellar rocket'),
-      },
-      {
-        name: 'Even more advanced options',
-        id: 'advanced2',
-        type: 'submenu',
-        action: (state: boolean) =>
-          console.log('Even more advanced options', state),
-        children: [
+interface MenuProps {
+  device: Device;
+}
+
+const fetchDeviceInfo = (device: Device) => {
+  return device.getDeviceInfo();
+};
+
+export const MenuComponent: Component<MenuProps> = ({ device }) => {
+  const [deviceInfo] = createResource(device, fetchDeviceInfo);
+
+  const header = (
+    <>
+      <h1>Castmill Player</h1>
+      <p>
+        Player: {deviceInfo()?.appType} {deviceInfo()?.appVersion}
+      </p>
+      <p>OS: {deviceInfo()?.os}</p>
+    </>
+  );
+
+  const capabilities = device.getCapabilities();
+
+  const entries: MenuEntry[] = [
```

Comment:
Maybe in the future there is a way to refactor the creation of this array to avoid code repetition.

Reply:
added a function for a more terse way of creating the menu entries